@here/harp-mapview-decoder/test/DataProviderTest.ts

```diff
@@ -60,4 +71,75 @@ describe("DataProvider", function() {
         dataProvider.unregister(clients[1]);
         expect(disposeSpy.calledOnce).is.true;
     });
+
+    it("calls connect only on first registered client", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+        const connectSpy = sinon.spy(dataProvider, "connect");
+
+        await dataProvider.register(clients[0]);
+        expect(connectSpy.calledOnce).is.true;
+
+        await dataProvider.register(clients[0]);
+        await dataProvider.register(clients[1]);
+        expect(connectSpy.calledOnce).is.true;
+    });
+
+    it("calls connect again after unregister", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+        const connectSpy = sinon.spy(dataProvider, "connect");
+
+        dataProvider.register(clients[0]);
+        expect(connectSpy.calledOnce).is.true;
+
+        dataProvider.unregister(clients[0]);
+        expect(connectSpy.calledOnce).is.true;
+
+        dataProvider.register(clients[0]);
+        expect(connectSpy.calledTwice).is.true;
+    });
+
+    it("calls connect after one provider call unregister", async function() {
+        const dataProvider = new TimedFakeDataProvider();
+
+        const promise0 = dataProvider.register(clients[0]);
+        const promise1 = dataProvider.register(clients[1]);
+        dataProvider.unregister(clients[0]);
+
+        let promise0Resolved = false;
+        promise0.then(() => {
```

Comment:
you should await on the `then` also

Reply:
or if you don't want to await on it now, store a reference to it then await on it when you need to

Reply:
done